File: src/main/java/com/wlyscraft/slimefunitemexpansion/machines/CompressedFreezer.java
package com.wlyscraft.slimefunitemexpansion.machines;

import io.github.thebusybiscuit.slimefun4.core.attributes.RecipeDisplayItem;
import io.github.thebusybiscuit.slimefun4.implementation.SlimefunItems;
import me.mrCookieSlime.Slimefun.Lists.RecipeType;
import me.mrCookieSlime.Slimefun.Objects.Category;
import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.abstractItems.AContainer;
import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.abstractItems.MachineRecipe;
import me.mrCookieSlime.Slimefun.api.SlimefunItemStack;
import org.bukkit.Material;
import org.bukkit.inventory.ItemStack;

import javax.annotation.Nonnull;
import java.util.ArrayList;
import java.util.List;

public class CompressedFreezer extends AContainer implements RecipeDisplayItem{
    public CompressedFreezer(Category category, SlimefunItemStack item, RecipeType recipeType, ItemStack[] recipe) {
        super(category, item, recipeType, recipe);
    }

    @Override
    protected void registerDefaultRecipes() {
        registerRecipe(15, new ItemStack[] { new ItemStack(Material.WATER_BUCKET) }, new ItemStack[] { new ItemStack(Material.BUCKET), new ItemStack(SlimefunItems.REACTOR_COOLANT_CELL) });
    }

    @Nonnull
    @Override
    public List<ItemStack> getDisplayRecipes() {
        List<ItemStack> displayRecipes = new ArrayList<>(recipes.size() * 2);

        for (MachineRecipe recipe : recipes) {
            displayRecipes.add(recipe.getInput()[0]);
            displayRecipes.add(recipe.getOutput()[recipe.getOutput().length - 1]);
        }

        return displayRecipes;
    }

    @Override
    public ItemStack getProgressBar() {
        return new ItemStack(Material.GOLDEN_PICKAXE);
    }

    @Nonnull
    @Override
    public String getMachineIdentifier() {
        return "IE_COMPRESSED_FREEZER";
    }

    @Override
    public int getEnergyConsumption() {
        // (J/s)
        return 60;
    }

    @Override
    public int getSpeed() {
        return 1;
    }

    @Override
    public int getCapacity() {
        // (J)
        return 512;
    }
}

File: src/main/kotlin/io/github/itsflicker/slimefunitemexpansion/items/machines/CompressedFreezer.kt
package io.github.itsflicker.slimefunitemexpansion.items.machines

import io.github.thebusybiscuit.slimefun4.api.items.ItemGroup
import io.github.thebusybiscuit.slimefun4.api.items.SlimefunItemStack
import io.github.thebusybiscuit.slimefun4.api.recipes.RecipeType
import io.github.thebusybiscuit.slimefun4.core.attributes.RecipeDisplayItem
import io.github.thebusybiscuit.slimefun4.implementation.SlimefunItems
import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.abstractItems.AContainer
import org.bukkit.Material
import org.bukkit.inventory.ItemStack

class CompressedFreezer(itemGroup: ItemGroup, item: SlimefunItemStack, recipeType: RecipeType, recipe: Array<ItemStack?>) :
    AContainer(itemGroup, item, recipeType, recipe), RecipeDisplayItem {

    override fun registerDefaultRecipes() {
        registerRecipe(
            20,
            arrayOf(ItemStack(Material.WATER_BUCKET)),
            arrayOf(ItemStack(Material.BUCKET), ItemStack(SlimefunItems.REACTOR_COOLANT_CELL))
        )
    }

    override fun getDisplayRecipes(): List<ItemStack> {
        val displayRecipes: MutableList<ItemStack> = ArrayList(recipes.size * 2)
        for (recipe in recipes) {
            displayRecipes.add(recipe.input[0])
            displayRecipes.add(recipe.output[recipe.output.size - 1])
        }
        return displayRecipes
    }

    override fun getProgressBar(): ItemStack {
        return ItemStack(Material.NETHERITE_PICKAXE)
    }

    override fun getMachineIdentifier(): String {
        return "IE_COMPRESSED_FREEZER"
    }
}